src/libraries/Microsoft.Extensions.Primitives/src/StringSegment.cs

```diff
@@ -434,21 +434,25 @@ public StringSegment Subsegment(int offset, int length)
         public int IndexOf(char c, int start, int count)
         {
             int offset = Offset + start;
+            int index = -1;
 
-            if (!HasValue || start < 0 || (uint)offset > (uint)Buffer.Length)
```

Comment:
>HasValue [](start = 17, length = 8)

I would say even if !HaveValue is true, we should still check the other parameters. i.e we should check start has to equal zero  (as the Offset should be zero too)

Reply:
I'm not sure that I understand what did you mean
if ```HasValue``` is ```false```, I'll check that ```start``` and ```offset``` is also zero (or not), and what should I do next?
How these values affect on returning value? If segment hasn't got value, the returning value will be definitely -1, am I missing something?

Reply:
what I meant is 

```C#
if (!HasValue && start != 0)
{
    // throw here
}
```

imagine some passing like someone passing `IndexOf('a', -1, x)` or passing ``IndexOf('a', 1_000_000, x)` . it is obviously wrong regardless of the string segment has a value or not. If you didn't throw, why we throw with similar values when the string segment has value then?

Reply:
Make sense